clippy_utils/src/ty.rs

```diff
@@ -538,13 +538,10 @@ pub fn same_type_and_consts<'tcx>(a: Ty<'tcx>, b: Ty<'tcx>) -> bool {
 }
 
 /// Checks if a given type looks safe to be uninitialized.
-pub fn is_uninit_value_valid_for_ty(cx: &LateContext<'_>, ty: Ty<'_>) -> bool {
-    match *ty.kind() {
-        ty::Array(component, _) => is_uninit_value_valid_for_ty(cx, component),
-        ty::Tuple(types) => types.iter().all(|ty| is_uninit_value_valid_for_ty(cx, ty)),
-        ty::Adt(adt, _) => cx.tcx.lang_items().maybe_uninit() == Some(adt.did()),
-        _ => false,
-    }
+pub fn is_uninit_value_valid_for_ty<'tcx>(cx: &LateContext<'tcx>, ty: Ty<'tcx>) -> bool {
+    cx.tcx
+        .check_validity_requirement((ValidityRequirement::Uninit, cx.param_env.and(ty)))
+        .unwrap_or(true)
```

Comment:
Should we be falling back to `false` here like before? I don't know the circumstances in which the function would return an `Err`

Reply:
It errors when it can't get the layout of the type, for example because it contains generic params. I chose true to avoid false positives, but it probably makes sense to make it false instead, so that clippy complains when using it on generic params (because that will probably be wrong).

Reply:
Thanks for explaining, yeah let's go with `false` in that case